future-apps/actor/contracts/Actor.sol

```diff
@@ -181,4 +178,28 @@ contract Actor is Vault, IERC165, IERC1271, IForwarder {
     function getSig(bytes data) internal pure returns (bytes4 sig) {
         assembly { sig := add(data, 0x20) }
     }
+
+    function addressAt(bytes data) internal pure returns (address) {
+        uint160 m = 0;
+        uint160 b = 0;
+
+        for (uint8 i = 0; i < 20; i++) {
+          m *= 256;
+          b = uint160(data[i]);
+          m += (b);
+        }
+
+        return address(m);
+    }
+
+    function bytesAfter(bytes data, uint offset) internal pure returns (bytes) {
+        uint length = data.length - offset;
+        bytes memory result = new bytes(length);
+
+        for (uint8 i = 0; i < length; i++) {
```

Comment:
This is terribly inefficient, you should use `memcpy`: https://github.com/Arachnid/solidity-stringutils/blob/master/src/strings.sol#L45